include/zephyr/kernel.h

```diff
@@ -4190,6 +4200,12 @@ struct k_work_q {
 
 	/* Flags describing queue state. */
 	uint32_t flags;
+
+#if defined(CONFIG_WORKQUEUE_WORK_TIMEOUT)
+	struct _timeout workto;
```

Comment:
please call it out, work_timeout or whatever

Reply:
that is exactly the problem, we have too many different "timeouts" and it is very confusing and makes the code difficult to read, if work_timeout is taken, then find something else that reflects what that is and the difference to  what is being used later.

The existing 'to' needs to be fixed, I did that in the past in many places, but few spots remain, but will come to that.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/77c2c454f70ceb9fb3e058e6523acccbe665cc76/kernel/timeout.c#L102

Reply:
I know and I am very well familiar with this issue, lets not make it worse please.

Reply:
_timeout is the timeout record (timeout_recod)
k_timeout_t is timeout type per the docs, so maybe call it this way, from looking at the code, everything is a timeout and it is very easy to mix things up.

Reply:
Sounds good :)

Reply:
Changed to `work_timeout_record` for the member and just `record` for the arg :)